lib/improver/utilities/cube_manipulation_new.py

```diff
@@ -314,367 +316,417 @@ def concatenate_cubes(
-            Merged cube.
+        Args:
+            cubelist (iris.cube.CubeList):
+                List of cubes to check and equalise.
 
-    """
-    if isinstance(cubes, iris.cube.Cube):
-        cubes = iris.cube.CubeList([cubes])
+        Returns:
+            cubelist (iris.cube.CubeList):
+                List of cubes with revised cubes. The number of cubes in this
+                list may be greater than the original number of cubes if they
+                have been sliced over mismatching dimension coordinates.
+        """
+        equalise_cube_attributes(cubelist, silent=self.silent_attributes)
+        strip_var_names(cubelist)
 
-    cubelist = _equalise_cubes(
-        cubes, model_id_attr=model_id_attr, merging=True)
+        cubelist = self._equalise_cube_coords(cubelist)
+        cubelist = self._equalise_cell_methods(cubelist)
+        for cube in cubelist:
+            check_cube_not_float64(cube, fix=True)  # TODO why here?
+        return cubelist
 
-    for i, cube in enumerate(cubelist):
-        cubelist[i] = iris.util.squeeze(cube)
+    def _equalise_cube_coords(self, cubes):
+        """
+        Function to equalise coordinates that do not match.
 
-    result = cubelist.merge_cube()
+        Args:
+            cubes (iris.cube.CubeList):
+                List of cubes to check the coords and revise.
 
-    if blend_coord is not None and blend_coord == "time":
-        # If bounds ranges did not match, "result" will not have a name
-        # on the leading dimension, but will have time and fp as aux
-        # coords.  Checking bounds ranges here raises a useful error,
-        # rather than encountering a missing blend coord downstream.
-        check_bounds_range_coords = ["time", "forecast_period"]
-        _check_bounds_ranges(result, check_bounds_range_coords)
+        Returns:
+            cubelist (iris.cube.CubeList):
+                List of cubes with revised coords. The number of cubes in this
+                list may be greater than the original number of cubes if they
+                have been sliced over mismatching dimension coordinates.
 
-    return result
+        Raises:
+            ValueError:
+                If coordinates in self.coord_mismatch_error_keys do not match.
+        """
+        # check for unmatching coords (returns a list of dictionaries, each
+        # with an entry for each mismatched coord.  If all coords match,
+        # returns a list of empty dictionaries.)
+        unmatching_coords = compare_coords(cubes)
+
+        # continue only if the list contains non-empty dictionaries
+        if bool(unmatching_coords[0]):
+
+            # check for mismatches in dim coord bounds
+            self._check_dim_coord_bounds(cubes)
+
+            # throw an error for specific coordinate mismatches
+            for error_key in self.coord_mismatch_error_keys:
+                # if necessary, translate threshold coordinate name
+                if error_key == "threshold":
```

Comment:
The new naming convention means that threshold is no longer called "threshold", but will have (for example) `standard_name="air_temperature"` and `var_name="threshold"`.  Since the mismatched keys will contain an actual coordinate standard or long name, we have to check the cubes for what the "threshold-type" coordinate is actually called.  This is one of the things incorporated from #846.

Reply:
How does this overlap with the change to the `self.coord_mismatch_error_keys` in the process method? If I'm reading the code right we'll never get into this loop because "threshold" will have been replaced in the `self.coord_mismatch_error_keys` the process method.

Reply:
This is an excellent point.  I will revert the behaviour in `_equalise_cube_coords` as I think that was what I originally intended to do.